### PR TITLE
databroker: support rotating shared secret

### PR DESCRIPTION
**Pull request by calebdoxsey**
## Summary
Support rotating the shared secret in the databroker by using a function to load the current shared key from the options instead of only doing so on first boot.

## Related issues
Fixes https://github.com/pomerium/internal/issues/928

## User Explanation
Support reloading more configuration options without a restart of the process.

## Checklist
- [x] reference any related issues
- [ ] updated docs
- [x] updated unit tests
- [ ] updated UPGRADING.md
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [x] ready for review

### CODE REVIEW

**Review by desimone — COMMENTED** [review]
LGTM pending @wasaga 's eyes

edit:

```
--- FAIL: TestNew (0.00s)
    --- FAIL: TestNew/bad_databroker_url (0.00s)
```

Looks real, albeit probably not a very meaningful test.

**Review by wasaga — APPROVED** [review]

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -118,20 +120,20 @@ jobs:
         idp: [auth0, azure, github, gitlab, google, oidc, okta, onelogin, ping]
     runs-on: ${{ matrix.platform }}
     steps:
-      - uses: actions/setup-go@84cbf8094393cdc5fe1fe1671ff2647332956b1a # pin@v2
+      - uses: actions/setup-go@84cbf8094393cdc5fe1fe1671ff2647332956b1a
```
Comment (desimone) [inline .github/workflows/test.yaml:123]: i think the linter gobbled the comments

File: databroker/cache.go [hunk 2]
```diff
@@ -36,11 +37,10 @@ type DataBroker struct {
 	manager          *manager.Manager
 	eventsMgr        *events.Manager
 
-	localListener                net.Listener
-	localGRPCServer              *grpc.Server
-	localGRPCConnection          *grpc.ClientConn
-	dataBrokerStorageType        string // TODO remove in v0.11
-	deprecatedCacheClusterDomain string // TODO: remove in v0.11
```
Comment (desimone) [inline databroker/cache.go:43]: did we remove all references to `dataBrokerStorageType` and `deprecatedCacheClusterDomain`; should we mention that in the changelog / upgrade guide or no?

**Review by desimone — APPROVED** [review]

File: databroker/cache.go (hunk 2 above)
Comment (calebdoxsey) [reply databroker/cache.go:43]: they were private. I'm not sure what could've referenced them.

File: .github/workflows/test.yaml (hunk 1 above)
Comment (calebdoxsey) [reply .github/workflows/test.yaml:123]: I removed the comments. They weren't accurate.

**Pull request merged**